android/java/org/chromium/chrome/browser/ntp/sponsored/SponsoredImageUtil.java

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.ntp.sponsored;
+
+import org.chromium.chrome.R;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.Calendar;
+
+public class SponsoredImageUtil {
```

Comment:
so it seems to me that `SponsoredImageUtil` should just be `SponsoredImage` and encapsulate all the details of generating an image for the NTP. When the NTP class is initialized it would call a static method on `SponsoredImage` to generate the `BitmapDrawable` and then the NTP would display it when `showBackgroundImage` is called. wdyt?

Reply:
or maybe a SponsoredImageView class to handle that?